### DISCUSSION TITLE
Adding clean support to zopen build

### DISCUSSION THREAD

**Discussion by perry-ca** [zopencommunity/meta#80] [Ideas]
I'd like to let everyone know that I added support to the `zopen build` command to allow you to clean up a build.  The zopen command is `zopen build -c` or `zopen build --cleanup`.  The default action for this will be to run `make clean`.  

If you need a different action, you can export the variables `ZOPEN_CLEAN` and `ZOPEN_CLEAN_OPTS` from the buildenv file.  For example ninjaport repo uses cmake.  The configure step sets up ../build as the build directory.    I added a clean up step using the following:
~~~
export ZOPEN_CLEAN=ninja_clean

ninja_clean()
{
 rm -rf ../build
}
~~~

You can also skip the clean up set by setting `ZOPEN_CLEAN=skip`.

After running (or skipping) the port's clean step, `zopen build -c` will delete any files it created in the previous build except for logs.  

The `zopen build -f` command now runs the clean step and then preforms the configure & build steps.

**Comment by IgorTodorovskiIBM** [comment]
Thanks @perry-ca !  I wonder if it makes sense to add the above content to the https://github.com/ZOSOpenTools/utils README.md or to https://zosopentools.github.io/meta/#/Guides/Porting.

**Comment by MikeFultonDev** [comment]
@IgorTodorovskiIBM and @perry-ca was this added? I'm marking it as reference. If it was added, can we close?

**Reply by IgorTodorovskiIBM** [reply]
Yes, all the remains is documentation, but if we're going to be using the --help output as the input for documentation then we can mark it closed.
